### ISSUE TITLE
Triplet Loss

### ISSUE THREAD

**Issue by prateekg** [swghosh/DeepFace#9]
Using accuracy metric and not using the triplet loss for learning the weights ??

**Comment by swghosh** [comment]
https://github.com/swghosh/DeepFace/issues/1#issuecomment-555747408

We're not using a triplet loss based implementation as this repo aims to replicate work done in the original [DeepFace paper](https://research.fb.com/publications/deepface-closing-the-gap-to-human-level-performance-in-face-verification/) which uses Softmax Classification for Face Recognition.

PS: [FaceNet](https://arxiv.org/abs/1503.03832) (a.k.a Triplet Loss) came much later to the publication of DeepFace. (a year later)

**Comment by prateekg** [comment]
Sure, but I think including the triplet loss will be a more enhanced and optimised version to implement the facial recognition system.

**Comment by swghosh** [comment]
Right. Triplet loss based models indeed have an edge over face recognition tasks. FaceNet and the alike can well support one shot learning and does not necessarily need to be trained again for the target recognition task.

Since, you're interested in Triplet Loss based models probably you could look into this repo: https://github.com/davidsandberg/facenet

And, in case you're willing to train your own FaceNet model from scratch you could refer to my other repo (which implements Triplet loss based FaceNet using the NN2 architecture in the paper): https://github.com/swghosh/FaceNet
